service/controller/resource/instance/create_wait_for_nodes_to_become_ready.go

```diff
@@ -29,6 +31,31 @@ func (r *Resource) waitForWorkersToBecomeReadyTransition(ctx context.Context, ob
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "found out that all tenant cluster worker nodes are Ready")
 
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	// If the old VMSS still exists, we want to go to a different state.
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Checking if the legacy VMSS %s is still present", key.LegacyWorkerVMSSName(cr))) // nolint: errcheck
+	vmss, err := r.getScaleSet(ctx, key.ResourceGroupName(cr), key.LegacyWorkerVMSSName(cr))
+	if IsScaleSetNotFound(err) {
+		return "", microerror.Mask(err)
```

Comment:
just making sure (you know best): this will restart the state machine , is that really what we want?

Reply:
Huh? No, it's returning error so that won't change the state.

Reply:
Sorry, I don't know what I was thinking.